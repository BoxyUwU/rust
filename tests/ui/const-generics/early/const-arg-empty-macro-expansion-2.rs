impl<T>
    Foo<
        //~^ ERROR: cannot find type `Foo` in this scope
        T,
        {
            thread_local! { pub static FOO : Foo = Foo { } ; }
            //~^ ERROR: cannot find type `Foo` in this scope
            //~| ERROR: cannot find type `Foo` in this scope
            //~| ERROR: cannot find type `Foo` in this scope
            //~| ERROR: cannot find type `Foo` in this scope
            //~| ERROR: cannot find type `Foo` in this scope
            //~| ERROR: cannot find struct, variant or union type `Foo` in this scope
        },
    >
{
}

fn main() {}
