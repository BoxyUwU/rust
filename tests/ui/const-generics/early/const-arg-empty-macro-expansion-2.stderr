error[E0412]: cannot find type `Foo` in this scope
  --> $DIR/const-arg-empty-macro-expansion-2.rs:2:5
   |
LL |     Foo<
   |     ^^^ not found in this scope

error[E0412]: cannot find type `Foo` in this scope
  --> $DIR/const-arg-empty-macro-expansion-2.rs:6:46
   |
LL |             thread_local! { pub static FOO : Foo = Foo { } ; }
   |                                              ^^^ not found in this scope

error[E0412]: cannot find type `Foo` in this scope
  --> $DIR/const-arg-empty-macro-expansion-2.rs:6:46
   |
LL |             thread_local! { pub static FOO : Foo = Foo { } ; }
   |                                              ^^^ not found in this scope
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error[E0422]: cannot find struct, variant or union type `Foo` in this scope
  --> $DIR/const-arg-empty-macro-expansion-2.rs:6:52
   |
LL |             thread_local! { pub static FOO : Foo = Foo { } ; }
   |                                                    ^^^ not found in this scope

error[E0412]: cannot find type `Foo` in this scope
  --> $DIR/const-arg-empty-macro-expansion-2.rs:6:46
   |
LL |             thread_local! { pub static FOO : Foo = Foo { } ; }
   |                                              ^^^ not found in this scope
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error[E0412]: cannot find type `Foo` in this scope
  --> $DIR/const-arg-empty-macro-expansion-2.rs:6:46
   |
LL |             thread_local! { pub static FOO : Foo = Foo { } ; }
   |                                              ^^^ not found in this scope
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error[E0412]: cannot find type `Foo` in this scope
  --> $DIR/const-arg-empty-macro-expansion-2.rs:6:46
   |
LL |             thread_local! { pub static FOO : Foo = Foo { } ; }
   |                                              ^^^ not found in this scope
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: aborting due to 7 previous errors

Some errors have detailed explanations: E0412, E0422.
For more information about an error, try `rustc --explain E0412`.
